hydra_provider/src/data_feed_service/serial/iterator.rs

```diff
@@ -0,0 +1,52 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use messages::mavlink::MavConnection;
+use postcard::from_bytes;
+use log::info;
+
+use crate::hydra_input::HydraInput;
+use messages::Message;
+use crate::data_feed_service::proto::SerialDataFeedConfig;
+
+use messages::mavlink::uorocketry::MavMessage;
+
+#[derive(Clone)]
+pub struct SerialDataFeedIterator {
+	pub config: Arc<Mutex<Option<SerialDataFeedConfig>>>,
+	pub mavlink: Arc<Mutex<Option<Box<dyn MavConnection<MavMessage> + Send + Sync>>>>,
+	pub is_running: Arc<AtomicBool>,
```

Comment:
Do those need to be behind a `Arc`? I don't see any clones on it, and it seems only `SerialDataFeedIterator` owns it.

Reply:
It's being accessed from a different async context within the service checkout `/serial/service.rs`. Since the iterator is accessing it to check if it should run and the service from a different async context might access it to change these fields.